fuzz/fuzz_targets/http_api.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright © 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use micro_http::Request;
+use once_cell::sync::Lazy;
+use std::os::unix::io::{AsRawFd, RawFd};
+use std::sync::mpsc::{channel, Receiver};
+use std::thread;
+use vmm::api::{http::*, ApiRequest, ApiResponsePayload};
+use vmm::{EpollContext, EpollDispatch};
+use vmm_sys_util::eventfd::EventFd;
+
+static ROUTES: Lazy<Vec<&Box<dyn EndpointHandler + Sync + Send>>> = Lazy::new(|| {
+    let mut keys: Vec<&String> = HTTP_ROUTES.routes.keys().collect();
+    keys.sort();
+    keys.iter()
+        .map(|k| HTTP_ROUTES.routes.get(*k).unwrap())
+        .collect()
+});
+
+fuzz_target!(|bytes| {
+    if bytes.len() < 2 {
+        return;
+    }
+
+    let route = ROUTES[bytes[0] as usize % ROUTES.len()];
+    if let Some(request) = generate_request(&bytes[1..]) {
+        let mut epoll = EpollContext::new().unwrap();
+        let exit_evt = EventFd::new(libc::EFD_NONBLOCK).unwrap();
+        let api_evt = EventFd::new(libc::EFD_NONBLOCK).unwrap();
+        epoll.add_event(&exit_evt, EpollDispatch::Exit).unwrap();
+        epoll.add_event(&api_evt, EpollDispatch::Api).unwrap();
+
+        let (api_sender, api_receiver) = channel();
+
+        let http_api_thread = {
+            let exit_evt = exit_evt.try_clone().unwrap();
+            let api_evt = api_evt.try_clone().unwrap();
+            thread::Builder::new()
+                .name("http_api".to_string())
+                .spawn(move || {
+                    route.handle_request(&request, api_evt, api_sender);
+                    exit_evt.write(1).ok();
+                })
+                .unwrap()
+        };
+
+        http_receiver_stub(epoll.as_raw_fd(), api_evt, api_receiver);
+        http_api_thread.join().unwrap();
```

Comment:
That's interesting how I would have taken the opposite approach of running the `http_receiver_stub` in a dedicated thread, and only once I'm sure it's running, I would have called into `route.handle_request()` from the main thread. It reads a bit better IMO since you want to run from the main thread the actual test.

Reply:
Good inputs. Updated. Also added the comments for why sorted `ROUTES` is required.